ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/DevMachine.java

```diff
@@ -42,29 +46,43 @@
 
     public DevMachine(@NotNull Machine devMachineDescriptor) {
         this.devMachineDescriptor = devMachineDescriptor;
+        this.machineConfig = devMachineDescriptor.getConfig();
         this.devMachineLinks = devMachineDescriptor instanceof Hyperlinks ? ((Hyperlinks)devMachineDescriptor).getLinks() : null;
 
         Map<String, ? extends Server> serverDtoMap = devMachineDescriptor.getRuntime().getServers();
         servers = new HashMap<>(serverDtoMap.size());
         for (String s : serverDtoMap.keySet()) {
             servers.put(s, new DevMachineServer(serverDtoMap.get(s)));
         }
-        runtimeProperties = devMachineDescriptor.getRuntime().getProperties();
-        envVariables = devMachineDescriptor.getRuntime().getEnvVariables();
+
+        MachineRuntimeInfo machineRuntime = devMachineDescriptor.getRuntime();
+        runtimeProperties = machineRuntime != null ? machineRuntime.getProperties() : null;
```

Comment:
@dkuleshov runtimeProperties is declared as final variable, so as I understand we need to assign a value or set null. Isn't it?